webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala

```diff
@@ -350,12 +349,40 @@ class ProcessFile(
    * a "file written" event.
    */
   type WatchQueue = ZQueue[Any, Nothing, Any, Nothing, WatchEvent, WatchEvent]
-  val queue = ZioRuntime.unsafeRun(Queue.bounded[WatchEvent](10000))
+  val watchEventQueue = ZioRuntime.unsafeRun(Queue.bounded[WatchEvent](16384))
+
+  /*
+   * We have a second queue that is used as buffer for the step between "file ok to be processed"
+   * and "saving file" because "saving file imply parsing XML and doing other memory and CPU
+   * inefficient.
+   * The buffer also allowed to filter inventory for identical nodes to avoid parsing old/not up to date
+   * inventory.
+   * As only a small structure is saved (SaveInventory is a name and two function to create stream from file), buffer can be big.
+   * When queue is full, we prefer to drop old inventories to new ones. In the worst case, they will be catched up by the
+   * scheduler.
```

Comment:
it can still be surprising. Why not using a bounded queue?

Reply:
I prefer to have a non blocking, RAM-bounbed structure. If there is a bug somewhere in the code (which is far from impossible for concurrent and async processing which involves inotify and moving files around. A blocking structure can lead to thread accumulation or even deadlock, while here, in the worst case we loose an event (and so an inventory processing). But it's easy to set up workaround for that, which is: `SchedulerMissedNotify`.